src/metabase/models/collection.clj

```diff
@@ -1032,9 +1032,8 @@
        ;; [{:name "A"
        ;;   :children [{:name "B"}, ...]}]
        ([m]
-        (let [vs (for [v (vals m)]
-                   (cond-> v
-                     (:children v) (update :children ->tree)))]
-          (sort-by (fn [{coll-name :name, coll-id :id}]
-                     [((fnil u/lower-case-en "") coll-name) coll-id]) vs))))
+        (->> (vals m)
+             (map #(update % :children ->tree))
+             (sort-by (fn [{coll-name :name, coll-id :id}]
+                        [((fnil u/lower-case-en "") coll-name) coll-id])))))
```

Comment:
Separate from the sketchiness of merging the entities, I do like the consistency of always including it. It's nice to be able to rely on having an array.